Build.cmake

```diff
@@ -93,20 +94,32 @@ ENDIF()
    file(GLOB SRC_FILES ${LOG_SRC}/g3log/*.h ${LOG_SRC}/g3log/*.hpp ${LOG_SRC}/*.cpp ${LOG_SRC}/*.ipp)
    file(GLOB HEADER_FILES ${LOG_SRC}/g3log/*.hpp ${LOG_SRC}/*.hpp)
    
+   list( APPEND HEADER_FILES ${GENERATED_G3_DEFINITIONS} )
+   list( APPEND SRC_FILES ${GENERATED_G3_DEFINITIONS} )
 
    IF (MSVC OR MINGW)
-         list(REMOVE_ITEM SRC_FILES  ${LOG_SRC}/crashhandler_unix.cpp)
+      list(REMOVE_ITEM SRC_FILES  ${LOG_SRC}/crashhandler_unix.cpp)
    ELSE()
-         list(REMOVE_ITEM SRC_FILES  ${LOG_SRC}/crashhandler_windows.cpp ${LOG_SRC}/g3log/stacktrace_windows.hpp ${LOG_SRC}/stacktrace_windows.cpp)
+      list(REMOVE_ITEM SRC_FILES  ${LOG_SRC}/crashhandler_windows.cpp ${LOG_SRC}/g3log/stacktrace_windows.hpp ${LOG_SRC}/stacktrace_windows.cpp)
    ENDIF (MSVC OR MINGW)
 
    set(SRC_FILES ${SRC_FILES} ${SRC_PLATFORM_SPECIFIC})
 
    # Create the g3log library
    INCLUDE_DIRECTORIES(${LOG_SRC})
-  SET(G3LOG_LIBRARY g3logger)
+   SET(G3LOG_LIBRARY g3logger)
+
+   IF( G3_SHARED_LIB )
+      IF(NOT(CMAKE_VERSION LESS 3.4))
+         set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+      ELSE()
+         message( FATAL "Need CMake version >=3.4 to build shared windows library!" )
+      ENDIF()
+      ADD_LIBRARY(${G3LOG_LIBRARY} SHARED ${SRC_FILES})
+   ELSE()
+      ADD_LIBRARY(${G3LOG_LIBRARY} STATIC ${SRC_FILES})
```

Comment:
The list APPEND is IMHO necessary, even on linux. If it did install, maybe it was also left over in the source directory and thus added through the file(GLOB command earlier? 
GLOB will not recursively add files, so even during an in-source build (in a build subdirectory as you do on linux) it should not get added.